activesupport/lib/active_support/message_encryptor.rb

```diff
@@ -115,7 +115,7 @@ def _decrypt(encrypted_message)
         # Currently the OpenSSL bindings do not raise an error if auth_tag is
         # truncated, which would allow an attacker to easily forge it. See
         # https://github.com/ruby/openssl/issues/63
-        raise InvalidMessage if aead_mode? && auth_tag.bytes.length != 16
+        raise InvalidMessage if aead_mode? && (auth_tag.nil? || auth_tag.bytes.length != 16)
```

Comment:
I created PR rails/rails/pull/29086 for this since we'll likely want to backport this fix to Rails 5.1 (which is when AEAD mode was introduced into `MessageEncryptor`.